homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -341,7 +411,189 @@ class PowerViewShade(PowerViewShadeBase):
     def _async_process_updated_position_data(self, position_data):
         """Process position data."""
         if ATTR_POSITION1 in position_data:
-            self._current_hd_cover_position = int(position_data[ATTR_POSITION1])
+            self._current_hd_cover_primary = int(position_data[ATTR_POSITION1])
+
+
+class PowerViewShadeTDBU(PowerViewShade):
+    """Representation of a PowerView shade with top/down bottom/up capabilities."""
+
+    @property
+    def current_cover_position_secondary(self):
+        """Return the current position of cover."""
+        return hd_position_to_hass(self._current_hd_cover_secondary)
+
+    async def async_open_cover(self, **kwargs):
+        """Close the cover."""
+        current_hass_pos1 = self.current_cover_position
+        current_hass_pos2 = self.current_cover_position_secondary
+        steps_to_move = 100 - (current_hass_pos1 + current_hass_pos2)
+        self._async_schedule_update_for_transition(steps_to_move)
+        self._async_update_from_command(await self._shade.open())
+
+    async def async_close_cover(self, **kwargs):
+        """Close the cover."""
+        current_hass_pos1 = self.current_cover_position
+        current_hass_pos2 = self.current_cover_position_secondary
+        steps_to_move = current_hass_pos1 + current_hass_pos2
+        self._async_schedule_update_for_transition(steps_to_move)
```

Comment:
It looks like this code and the code on 427 could be made more DRY and moved into a new function which takes 100 as an argument